src/main/java/com/adobe/aem/compgenerator/web/PropertyBuilderServlet.java

```diff
@@ -0,0 +1,285 @@
+        }
+        if (reConfig.has("useExistingModel")) {
+            property.setUseExistingModel(reConfig.get("useExistingModel").asBoolean());
+        }
+        if (reConfig.has("jsonProperty")) {
+            property.setJsonProperty(reConfig.get("jsonProperty").asText());
+        }
+        if (reConfig.has("modelName")) {
+            property.setModelName(reConfig.get("modelName").asText());
+        }
+        if (reConfig.has("type")) {
+            JsonNode type = reConfig.get("type");
+            String value = type.get("value").textValue();
+            property.setType(value);
+        }
+        if (reConfig.has("attributes")) {
+            Map<String, String> updatedAttributes = new HashMap<>();
+            JsonNode attrArr = reConfig.get("attributes");
+            if (attrArr.isArray()) {
+                for (final JsonNode objNode : attrArr) {
+                    updatedAttributes.put(objNode.get("key").asText(), objNode.get("value").asText());
+                }
+            }
+            property.setAttributes(updatedAttributes);
+        }
+        return property;
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        resp.setContentType("application/json; charset=UTF-8");
+        PrintWriter writer = resp.getWriter();
+        ObjectMapper mapper = new ObjectMapper();
+        Message msg = new Message(true, "Tab builder getter");
+        writer.write(mapper.writeValueAsString(msg));
+        writer.close();
+    }
+
+    @Override
+    /*
+     * Handles updates to the dialog properties configuration
+     */
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        resp.setContentType("application/json; charset=UTF-8");
+        resp.setCharacterEncoding("UTF-8");
+        ObjectMapper mapper = new ObjectMapper();
+        File configFile = new File(CONFIG_PATH);
+        // sanity check for config file existence
+        if (!configFile.exists()) {
+            resp.setStatus(500);
+            PrintWriter writer = resp.getWriter();
+            Message msg = new Message(false, MISSING_CONFIG_MSG);
+            writer.write(mapper.writeValueAsString(msg));
+            writer.close();
+            return;
+        }
+        boolean updated = false;
+        GenerationConfig config = CommonUtils.getComponentData(configFile);
+        Options options = config.getOptions();
+        // sanity check on request payload
+        if (req.getContentLength() <= 0) {
+            resp.setStatus(500);
+        } else {
+            // get config params as json from request body
+            String body = req.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
+            JsonNode reConfig = mapper.readTree(body);
+            if (reConfig.has("moveProp")) {
```

Comment:
I think it could help code readability a lot to move the contents of these `if` conditions to functions - e.g. a `moveProp()` function - it will leave the decision logic clear and drastically shorten `doPost()` to a more digestible format.

Reply:
Good idea. Will update this as well.